app/controllers/api/v1/pages_controller.rb

```diff
@@ -0,0 +1,17 @@
+module Api
+  module V1
+    class PagesController < ApiController
+      before_action :authenticate!, except: %i[show index]
```

Comment:
(question): The rest of my comments are based on the assumption above that we do not need to authenticate for index and show. 

However, it feels strange to me that these endpoints do not require an api key to access. Is this because we need the correct roles to access the api endpoint anyways?

Reply:
I'm not sure. Looking at other API endpoints, it seems we generally allow unauthenticated access to endpoints which present "public" information (show, index, etc.) and only restrict access for endpoints where it's possible to alter data (create, update, unpublish, suspend, etc.) At least one endpoint has an optional form of authentication, so `current_user` would be optionally set when an API key is used, but would not be required.

@fdocr I think you have the most experience with the V1 API so far, any chance you could help clarify this?.

Reply:
> Looking at other API endpoints, it seems we generally allow unauthenticated access to endpoints which present "public" information (show, index, etc.) and only restrict access for endpoints where it's possible to alter data (create, update, unpublish, suspend, etc.) At least one endpoint has an optional form of authentication, so current_user would be optionally set when an API key is used, but would not be required.

This point of view is sound in a general sense. I can see it going either way because once pages are created they are public (they can be visited from a browser as unauthenticated users).

However, managing pages is something that only exists in the admin dashboard and making these two endpoints public would _very slightly_ change this (being able to see the pages that exist and their config details) even with admin restricted access to create/update/destroy.

I'm totally okay with keeping it this way, but just a heads up to @pkfrank / @benhalpern about this, since from a product standpoint there's a chance we could prefer to keep all of these "admin only" regardless of the above.